x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java

```diff
@@ -683,8 +683,20 @@ private Settings getHttpTransportSSLSettings(Settings settings) {
         }
 
         Settings.Builder builder = Settings.builder().put(httpSSLSettings);
-        if (builder.get("client_authentication") == null) {
-            builder.put("client_authentication", XPackSettings.HTTP_CLIENT_AUTH_DEFAULT);
+        if (builder.get("client_authentication") == null && builder.get("verification_mode") == null) {
+            // by default, Security HTTPS does not validate the client
+            builder.put("client_authentication", SSLClientAuth.NONE);
+            builder.put("verification_mode", VerificationMode.NONE);
+        } else if (builder.get("client_authentication") == null) {
+            // otherwise it defaults to 'required' which is incompatible with `verification_mode` 'none`
+            if (VerificationMode.parse(builder.get("verification_mode")).equals(VerificationMode.NONE)) {
+                builder.put("client_authentication", SSLClientAuth.NONE);
+            }
```

Comment:
Is this correct?

If I have this config:
```
xpack.security.http.ssl:
  enabled: true
  keystore.path: keys.p12
  verification_mode: certificate
```

That would previously result in a somewhat non-sensical SSL config of:
- client-auth: none
- verification-mode: certificate
- truststore: JDK

But (assuming I'm reading this correctly) it would now give:
- client-auth: required
- verification-mode: certificate
- truststore: JDK

Reply:
You're right. And the new proposed behavior is wrong. Instead it should result in a deprecation warning (you only change the verification mode but the client auth is defaulting to none). There was also a test case with this exact example that I added...
I don't know what I was thinking, I was probably mesmerized by the symmetry of the condition.